### PR TITLE
Add promise handler

### PR DESCRIPTION
**Pull request by danfinlay**
Pretty simple change, adds the ability for `engine.handle(req, cb)` to also work as `await engine.handle(req)`. Made it because I wanted to use `await` when writing a dependent permission for rpc-cap.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -19,7 +19,30 @@ class RpcEngine extends SafeEventEmitter {
...
+        if (err) return reject(err)
+        return resolve(result)
+      }
+
+      // batch request support
+      if (Array.isArray(req)) {
+        this._handleBatch(req, callback)
+      } else {
+        this._handle(req, callback)
+      }
```
Comment (rekmarks) [inline src/index.js:41]: This should probably just call `handleCallback`.

**Review by rekmarks — CHANGES_REQUESTED** [review]
This is a good idea. There's but one minor change (see inline), otherwise looks good!

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-v10
+v13
```
Comment (whymarrh) [inline .nvmrc:1]: This isn't a LTS version, I would advise against using it right now

**Review by whymarrh — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (kumavis) [reply src/index.js:41]: yeah use pify on the handleCallback
```js
return pify(cb => this.handleCallback(req, cb)()
```

**Pull request closed without merging**